mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_token.go

```diff
@@ -0,0 +1,143 @@
+				// Validators: verify.ValidateRegexp("(" + strings.Join(verify.PossibleServiceAccountNames, "|") + ")"),
+			},
+			"access_token": schema.StringAttribute{
+				Sensitive: true,
+				Computed:  true,
+			},
+			"lifetime": schema.StringAttribute{
+				Optional: true,
+				// Validators: verify.ValidateDuration(), // duration <=3600s; TODO: support validateDuration(min,max)
+				// Default:      "3600s",
+			},
+			"scopes": schema.SetAttribute{
+				Required:    true,
+				ElementType: types.StringType,
+				// ValidateFunc is not yet supported on lists or sets.
+			},
+			"delegates": schema.SetAttribute{
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+		},
+	}
+}
+
+func (p *googleEphemeralServiceAccountAccessToken) Configure(ctx context.Context, req ephemeral.ConfigureRequest, resp *ephemeral.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	pd, ok := req.ProviderData.(*fwtransport.FrameworkProviderConfig)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *fwtransport.FrameworkProviderConfig, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	// Required for accessing userAgent and passing as an argument into a util function
+	p.providerConfig = pd
+}
+
+func (p *googleEphemeralServiceAccountAccessToken) Open(ctx context.Context, req ephemeral.OpenRequest, resp *ephemeral.OpenResponse) {
+	var data ephemeralServiceAccountAccessTokenModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if data.Lifetime.IsNull() {
+		data.Lifetime = types.StringValue("3600s")
+	}
+
+	service := p.providerConfig.NewIamCredentialsClient(p.providerConfig.UserAgent)
+	name := fmt.Sprintf("projects/-/serviceAccounts/%s", data.TargetServiceAccount.ValueString())
+
+	ScopesSetValue, diags := data.Scopes.ToSetValue(ctx)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var delegates []string
+	if !data.Delegates.IsNull() {
+		DelegatesSetValue, diags := data.Delegates.ToSetValue(ctx)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		delegates = StringSet(DelegatesSetValue)
+	}
+
+	tokenRequest := &iamcredentials.GenerateAccessTokenRequest{
+		Lifetime:  data.Lifetime.ValueString(),
+		Delegates: delegates,
+		Scope:     tpgresource.CanonicalizeServiceScopes(StringSet(ScopesSetValue)),
+	}
+
+	at, err := service.Projects.ServiceAccounts.GenerateAccessToken(name, tokenRequest).Do()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error generating access token",
+			fmt.Sprintf("Error generating access token: %s", err),
+		)
+		return
+	}
+
+	data.AccessToken = types.StringValue(at.AccessToken)
+	resp.Diagnostics.Append(resp.Result.Set(ctx, data)...)
+}
+
+func StringSet(d basetypes.SetValue) []string {
```

Comment:
this can be taken out and put in a utils file.

Reply:
this will be removed once https://github.com/GoogleCloudPlatform/magic-modules/pull/12295 is merged.

Reply:
resolved since it's been removed after #12295 was merged.